docs/BestPractices.md

```diff
@@ -204,3 +205,41 @@ FROM node:alpine as app
 ## Copy built node modules and binaries without including the toolchain
 COPY --from=builder node_modules .
 ```
+
+
+## Smaller images without npm/yarn
+
+If you want to achieve an even smaller image size than the `-alpine`, you can omit the npm/yarn like this:
+
+```Dockerfile
+FROM node:18-alpine3.16 AS builder
+WORKDIR /build-stage
+COPY package*.json ./
+RUN npm ci
+
+# Copy the the files you need
+COPY . ./
+
+RUN npm run build
+
+
+# Make sure the alpine version is the same as in the build stage
+FROM alpine:3.16
+RUN apk add --no-cache libstdc++ dumb-init
+RUN addgroup -g 1000 node && adduser -u 1000 -G node -s /bin/sh -D node
+
+COPY --from=builder /usr/local/bin/node /usr/local/bin/
+COPY --from=builder /usr/local/bin/docker-entrypoint.sh /usr/local/bin/
+ENTRYPOINT ["docker-entrypoint.sh"]
+
+# Update the following lines based on your codebase
+COPY --from=builder /build-stage/node_modules ./node_modules
+COPY --from=builder /build-stage/dist ./dist
+
+RUN chown -R node:node ./
```

Comment:
Missing a `WORKDIR` before the two `COPY` lines, otherwise this is `chown`ing much more than needed (and doubling the final image size).

I'd also just drop this in favor of just doing the `chown` during that `COPY` (([`COPY --chown=`](https://docs.docker.com/engine/reference/builder/#copy))) and thus avoid the double size of `node_modules` and `dist` in the final image (once for the `COPY` layer and once for the `RUN chown` layer).

Reply:
Thank you @yosifkit,

sorry my the mistake. My config is different, so I created the example just from head, without actually testing it. I have updated it following your suggestions. I had to add chown at the beginning since if the code is creating file in the workdir, it needs to have a permissions to do that.